### ISSUE TITLE
Possible to hook into addListItems?

### ISSUE THREAD

**Issue by kylebarron** [burodepeper/language-markdown#222]
Thanks again for the great package.

I'm working on another Atom package for Stata that does dynamic-document generation like R Markdown. Most of the non-highlighting capabilities of this package are defined as commands, like `markdown:indent-list-item` and `markdown:outdent-list-item`, etc, and they can be easily called from another package by assigning a key to those in the `keymap.cson` file. 

However for `addListItems`, since it's not in the `addCommands()` function, is there a way to easily call that from another program, without just copying and pasting the code?

**Comment by kylebarron** [comment]
Even copying and pasting `lib/main.js` and changing the `grammar.name` statement, somehow I couldn't recreate the `addListItems` functionality in `language-stata`.
https://github.com/burodepeper/language-markdown/blob/4424576d93037930b74d0a7b9c2557b22eb6555a/lib/main.js#L82

**Comment by burodepeper** [comment]
> However for addListItems, since it's not in the addCommands() function, is there a way to easily call that from another program, without just copying and pasting the code?

Hmm, I don't think so. This bit was kind of hacked together, as the code is pretending to type in the editor, instead of modifying the buffer. Because it is pretending to type, it automatically indents to the correct point, etc. I spend a little time on turning it into a proper command, but it just wasn't worth the effort at that point.

What exactly do you want to do with the function in your package? The same thing?

**Comment by kylebarron** [comment]
Yeah, I basically want to inject Stata syntax highlighting in between specific tags, but otherwise use all the functionality of the language-markdown package. For example, my file in `grammars/stata-dyndoc-md.cson` is pretty much
```cson
'patterns': [
  {
    'include': 'source.dyndoc.stata'
  }
  {
    'include': 'text.md'
  }
]
```

where `source.dyndoc.stata` is a set of rules for when to highlight Stata code. And the standard keymaps imitate the language-markdown keymaps:
```cson
'atom-text-editor[data-grammar="source dyndoc md stata"]':
  'tab': 'markdown:indent-list-item'
  'shift-tab': 'markdown:outdent-list-item'
  '_': 'markdown:emphasis'
  '*': 'markdown:strong-emphasis'
  '~': 'markdown:strike-through'
  '@': 'markdown:link'
  '!': 'markdown:image'
```

I don't particularly want to submit a PR for `language-markdown` with the changes, because I don't want to assign any responsibility for upkeep of the package.

It's not that big of a deal for `addListItems` to not work.

This would also be useful for language-pweave, a package for dynamic documents in Python, and might be what @mpastell was talking about in https://github.com/burodepeper/language-markdown/issues/209.

**Comment by burodepeper** [comment]
> Even copying and pasting lib/main.js and changing the grammar.name statement, somehow I couldn't recreate the addListItems functionality in language-stata.

Did you also include the `isEmbeddedCode()` method? That's the one thing I can imagine you might've skimmed over. I don't quickly see anything else in there that might be the culprit.

If only I had more time on my hands to make it into a proper standalone solution ; )

Anyway, if you can set me up with a branch with your copy/pasted work-in-progress and whatever I need (to know) about Stata (never heard of it before), I can have a quick look at it.

**Comment by burodepeper** [comment]
Any luck on this yet?
